src/test/java/com/google/cloud/spanner/pgadapter/statements/DdlExecutorTest.java

```diff
@@ -174,19 +174,37 @@ ImmutableList<Statement> getDropSchemaViewsStatements(TableOrIndexName schemaNam
         ImmutableList.of(
             Statement.of("drop dependent indexes of foo"), Statement.of("drop table foo")),
         ddlExecutor.getDependentStatements(Statement.of("drop table foo")));
+    assertEquals(
+        ImmutableList.of(
+            Statement.of("drop dependent indexes of foo"),
+            Statement.of("drop table if exists foo")),
+        ddlExecutor.getDependentStatements(Statement.of("drop table if exists foo")));
     assertEquals(
         ImmutableList.of(
             Statement.of("drop dependent indexes of foo"),
             Statement.of("drop dependent foreign keys of foo"),
             Statement.of("drop table foo")),
         ddlExecutor.getDependentStatements(Statement.of("drop table foo cascade")));
+    assertEquals(
+        ImmutableList.of(
+            Statement.of("drop dependent indexes of foo"),
+            Statement.of("drop dependent foreign keys of foo"),
+            Statement.of("drop table if exists foo")),
+        ddlExecutor.getDependentStatements(Statement.of("drop table if exists foo cascade")));
     assertEquals(
         ImmutableList.of(
             Statement.of("drop all indexes in schema foo"),
             Statement.of("drop all foreign keys in schema foo"),
             Statement.of("drop all views in schema foo"),
             Statement.of("drop all tables in schema foo")),
         ddlExecutor.getDependentStatements(Statement.of("drop schema foo cascade")));
+    assertEquals(
+        ImmutableList.of(
+            Statement.of("drop all indexes in schema foo"),
+            Statement.of("drop all foreign keys in schema foo"),
+            Statement.of("drop all views in schema foo"),
+            Statement.of("drop all tables in schema foo")),
+        ddlExecutor.getDependentStatements(Statement.of("drop schema if exists foo cascade")));
```

Comment:
If the table/schema doesn't exist, we will still return an error. This doesn't match the actual behaviour, right?

Reply:
No, the original statement is still the one that is being executed. So even though the parser eats the `if exists` clause, that is only to determine whether the statement contains a `cascade` option, and to find the table name that we need to use to search for dependent objects. The parser is not literally used to execute the original drop statement, which remains `drop schema|table if exists ....`.

See also the additional tests that were added based on the comments above.